homeassistant/components/group/binary_sensor.py

```diff
@@ -90,24 +90,21 @@ def __init__(
     async def async_added_to_hass(self) -> None:
         """Register callbacks."""
 
-        async def async_state_changed_listener(event: Event) -> None:
+        @callback
+        def async_state_changed_listener(event: Event) -> None:
             """Handle child updates."""
             self.async_set_context(event.context)
-            await self.async_defer_or_update_ha_state()
+            self.async_defer_or_update_ha_state()
 
         self.async_on_remove(
             async_track_state_change_event(
                 self.hass, self._entity_ids, async_state_changed_listener
             )
         )
 
-        if self.hass.state == CoreState.running:
-            await self.async_update()
-            return
-
```

Comment:
The superclass' `async_added_to_hass` also calls `async_update`, I think we can just use that?
Same for other components.